plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployCommandHandler.java

```diff
@@ -38,14 +39,8 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
     try {
       IProject project = helper.getProject(event);
       if (project != null) {
-        String now = Long.toString(System.currentTimeMillis());
-        StandardDeployJob deploy =
-            new StandardDeployJob(new ExplodedWarPublisher(),
-                                  new StandardProjectStaging(),
-                                  new AppEngineProjectDeployer(),
-                                  getTempDir().append(now),
-                                  project);
-        deploy.schedule();
+        launchDeployJob(project);
+        launchCleanupJob();
```

Comment:
I don't think this is right — you should only schedule the CleanupJob at the end of your DeployJob.  Because of your keeping the last two, it shouldn't have that much practical effect, but if you knock `OLD_DIRECTORIES_TO_KEEP` down to 0 you'll likely encounter interference between the jobs.

WTP has a nice (unfortunately internal) class `ChainedJob`.  We could create a more flexible utility class that supports chaining a set of arbitrary jobs through job listeners.